packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -3440,8 +3495,28 @@ class InputDecorationTheme extends Diagnosticable {
   /// the input has focus or text has been entered.
   ///
   /// Defaults to true.
+  @Deprecated(
+    'Use floatLabelBehaviour instead. '
+    'This feature was deprecated after v1.13.2.'
+  )
   final bool hasFloatingPlaceholder;
 
+  /// Defines how the floating label should be displayed.
+  ///
+  /// When [FloatLabelBehavior.auto] the label will float to the top only when
+  /// the field is focused or has some text content, otherwise it will appear
+  /// in the field in place of the content.
+  ///
+  /// When [FloatLabelBehavior.always] the label will always float at the top
+  /// of the field above the content.
+  ///
+  /// When [FloatLabelBehavior.never] the label will always will appear
+  /// in the field in place of the content.
+  ///
+  /// Defaults to [FloatLabelBehavior.auto].
+  ///
```

Comment:
You can use a macro to avoid duplicating this comment.  Above, where you first document floatLabelBehavior, wrap it in a template comment like this:

```
/// {@template flutter.material.inputDecoration.floatLabelBehavior}
/// ... your documentation
/// {@endtemplate}
```

Then you can replace all of the duplicate docs here with:

```
/// {@macro flutter.material.inputDecoration.floatLabelBehavior}
```

Reply:
Oh neat!